### ISSUE TITLE
consider removing support for older rubies

### ISSUE THREAD

**Issue by flavorjones** [flavorjones/mini_portile#123]
It's getting hard to support old rubies, see failing test suites here related to setup-ruby and bundler versions:

https://github.com/flavorjones/mini_portile/actions/runs/3771621986/jobs/6412111557

I'd like to consider dropping support for ruby < 2.6 in the next minor release.

**Comment by halostatue** [comment]
Still worth considering. `minitar` v1 has been released and only supports Ruby 3.1 or later. There shouldn't be any other changes required since you are not using the `Archive::Tar` namespace.

**Comment by flavorjones** [comment]
@halostatue Thanks for that context! I've only not-removed support because it hasn't bitten me since I opened this ticket. :four_leaf_clover: It's only a matter of "when" we drop support, not "if". :heart:

**Comment by halostatue** [comment]
Yeah. I’m getting ready to go through a similar upgrade process for most of my maintained gems, although I probably won't be cutting them "hard" like I have done with minitar (that is, when I cut mime-types 4, I will continue to maintain mime-types 3 for some time to come, but new features will not show up in mime-types 3; same with diff-lcs 2 vs diff-lcs 1).

I need to open a discussion on ruby-core to get some advice from yjit developers so I can examine these for ways to ensure that these gems are easier to optimize.
